src/order/prime_ideal.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2021 Noam Atar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Noam Atar
+-/
+import order.basic
+import order.ideal
+import order.pfilter
+
+/-!
+# Prime ideals
+
+## Main definitions
+
+Throughout this file, `P` is at least a preorder, but some sections require more
+structure, such as a bottom element, a top element, or a join-semilattice structure.
+
+- `order.ideal.prime_pair`: A pair of an `ideal` and a `pfilter` which form a partition of `P`.
+  This is useful as giving the data of a prime ideal is the same as giving the data of a prime
+  filter.
+- `order.ideal.is_prime`: a predicate for prime ideals. Dual to the notion of a prime filter.
+- `order.pfilter.is_prime`: a predicate for prime filters. Dual to the notion of a prime ideal.
+
+## References
+
+- <https://en.wikipedia.org/wiki/Ideal_(order_theory)>
+
+## Tags
+
+ideal, prime
+
+-/
+
+namespace order
+
+variables {P : Type*}
+
+section preorder
+
+variables [preorder P]
+
+namespace ideal
+
+/-- A pair of an `ideal` and a `pfilter` which form a partition of `P`.
+-/
+@[nolint has_inhabited_instance]
+structure prime_pair (P : Type*) [preorder P] :=
+(I            : ideal P)
+(F            : pfilter P)
+(compl_I_eq_F : (I : set P)ᶜ = F)
```

Comment:
would this be better stated this way?
```suggestion
(is_compl_I_F : is_compl I F)
```

Reply:
Oh yeah, this might be even better! Let me give it a shot.